Bug: foo instead of fixme

F
======================================================================
FAIL: test_foo (tests.test_gilded_rose.GildedRoseTest.test_foo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/opt/code/learning-test-driven-development-gilded-rose-kata/python/tests/test_gilded_rose.py", line 12, in test_foo
    self.assertEqual("fixme", items[0].name)
AssertionError: 'fixme' != 'foo'
- fixme
+ foo


----------------------------------------------------------------------
Ran 1 test in 0.002s

FAILED (failures=1)

--- a/300/100/200/100/100/README.md
+++ b/300/100/200/100/100/README.md
@@ -1,3 +1,88 @@
 # 100 - Work on your fix: "fixme instead of foo"
 
-TO DO
+## Finding the Issue: 'foo found but fixme expected'
+
+### Step 1: Run the failing test to see the error
+
+```bash
+# Make sure you're in the python directory with your virtual environment activated
+cd /path/to/learning-test-driven-development-gilded-rose-kata/python
+source ../venv/bin/activate  # or wherever your venv is located
+
+# Run the unit tests to see the failure
+python -m unittest
+```
+
+### Step 2: Analyze the test failure output
+
+You should see output similar to:
+```
+F
+======================================================================
+FAIL: test_foo (tests.test_gilded_rose.GildedRoseTest.test_foo)
+----------------------------------------------------------------------
+Traceback (most recent call last):
+  File "/path/to/tests/test_gilded_rose.py", line 12, in test_foo
+    self.assertEqual("fixme", items[0].name)
+AssertionError: 'fixme' != 'foo'
+- fixme
++ foo
+
+----------------------------------------------------------------------
+Ran 1 test in 0.001s
+
+FAILED (failures=1)
+```
+
+### Step 3: Examine the failing test
+
+```bash
+# Look at the test file to understand what it expects
+cat tests/test_gilded_rose.py
+```
+
+Look for the `test_foo` method. It should contain something like:
+```python
+def test_foo(self):
+    items = [Item("foo", 0, 0)]
+    gilded_rose = GildedRose(items)
+    gilded_rose.update_quality()
+    self.assertEqual("fixme", items[0].name)  # This line is failing
+```
+
+### Step 4: Examine the current implementation
+
+```bash
+# Look at the gilded_rose.py file to see what's actually happening
+cat gilded_rose.py
+```
+
+### Step 5: Identify the problem
+
+The issue is that:
+- **Test expects**: The item name should be "fixme" after processing
+- **Current behavior**: The item name remains "foo" 
+- **Root cause**: The `update_quality()` method in `GildedRose` class is not changing the item name from "foo" to "fixme"
+
+### Step 6: Locate the specific code that needs fixing
+
+Look for the `update_quality` method in `gilded_rose.py`. The problem is likely in this method where it should be changing the item name but isn't.
+
+### Step 7: Understand the business logic
+
+Based on the test, the expected behavior is:
+1. Create an item with name "foo"
+2. Process it through `GildedRose.update_quality()`
+3. The item name should change to "fixme"
+
+### Step 8: Verify your understanding
+
+```bash
+# Run the test again to confirm the issue
+python -m unittest tests.test_gilded_rose.GildedRoseTest.test_foo -v
+```
+
+This will show you exactly which test is failing and what the expected vs actual values are.
+
+
+TO DO: FIX IT

--- a/300/100/200/100/100/README.md
+++ b/300/100/200/100/100/README.md
@@ -85,4 +85,46 @@ python -m unittest tests.test_gilded_rose.GildedRoseTest.test_foo -v
 This will show you exactly which test is failing and what the expected vs actual values are.
 
 
-TO DO: FIX IT
+## Fixing the Issue: 'foo found but fixme expected'
+
+In fact the error is not in de application (here: ```gilded_rose.py```), but in the unit test itself (here: ```test_gilded_rose.py```) in the test ```test_foo```:
+
+```
+    def test_foo(self):
+        items = [Item("foo", 0, 0)]
+        gilded_rose = GildedRose(items)
+        gilded_rose.update_quality()
+        self.assertEqual("fixme", items[0].name)
+```
+
+It can be easily fixed as follows:
+
+```
+    def test_foo(self):
+        items = [Item("foo", 0, 0)]
+        gilded_rose = GildedRose(items)
+        gilded_rose.update_quality()
+        self.assertEqual("foo", items[0].name) # changed "fixme" to "foo"
+```
+
+When running the unit tests again, the error has gone:
+
+``` python
+python -m unittest
+```
+
+The outcome is:
+
+```
+.
+----------------------------------------------------------------------
+Ran 1 test in 0.000s
+
+OK
+```
+
+The test was **intentional** faulty, so we would have an idea of the unit tests running well - it’s just showing you how the test structure works.
+
+Next step is to create additional unit tests that cover the application's logic to fullfill all requirements set at the start. This we will pick up in their own section.
+
+Now we can commit the change back to the branch and start a pull request for merging with the main branch.

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -66,7 +66,7 @@ Your branch is up to date with 'origin/main'.
 nothing to commit, working tree clean
 ```
 
-## Method 1: Local branch creation for bug fixing
+## Method 1: Local branch creation for bug fixing (Not Recommended)
 
 ### Create a new branch locally
 

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -125,7 +125,180 @@ See [README.md](./100/README.md)
 ```
 git add .
 git commit -m "Fix: Change item name from 'foo' to 'fixme'"
-git push origin bug/foo-instead-of-fixme
+git push origin bug/2-bug-foo-instead-of-fixme
 ```
 
-MORE
+After pushing your branch, you can create a Pull Request (PR) in several ways. Here are the most common methods:
+
+## Method 1: GitHub Web Interface (Recommended)
+1. Go to your repository on GitHub
+2. You'll see a yellow banner saying "bug/2-bug-foo-instead-of-fixme had recent pushes" with a "Compare & pull request" button
+3. Click "Compare & pull request"
+4. Fill out the PR form:
+- Title: "Fix: Change item name from 'foo' to 'fixme'"
+- Description: Explain what the bug was and how you fixed it
+- Reviewers: Add any team members who should review
+5.Click "Create pull request"
+
+## Method 2: GitHub CLI (if you have it installed)
+
+### Installing GitHub CLI
+
+#### macOS (using Homebrew)
+```bash
+# Install Homebrew if you don't have it
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+# Install GitHub CLI
+brew install gh
+```
+
+#### macOS (using MacPorts)
+```bash
+sudo port install gh
+```
+
+#### Windows (using Chocolatey)
+```powershell
+# Install Chocolatey if you don't have it
+Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
+
+# Install GitHub CLI
+choco install gh
+```
+
+#### Windows (using Scoop)
+```powershell
+# Install Scoop if you don't have it
+Set-ExecutionPolicy RemoteSigned -Scope CurrentUser
+irm get.scoop.sh | iex
+
+# Install GitHub CLI
+scoop install gh
+```
+
+#### Linux (Ubuntu/Debian)
+```bash
+# Add GitHub CLI repository
+curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+
+# Update package list and install
+sudo apt update
+sudo apt install gh
+```
+
+#### Linux (Fedora/RHEL/CentOS)
+```bash
+# Add GitHub CLI repository
+sudo dnf install 'https://github.com/cli/cli/releases/download/v2.40.1/gh_2.40.1_linux_amd64.rpm'
+```
+
+#### Manual Installation (All Platforms)
+```bash
+# Download the latest release from GitHub
+# Visit: https://github.com/cli/cli/releases/latest
+# Download the appropriate package for your OS and architecture
+```
+
+### Authenticating with GitHub CLI
+
+```bash
+# Authenticate with GitHub (opens browser for login)
+gh auth login
+
+# Choose your preferred protocol (HTTPS recommended)
+# Choose your preferred authentication method (Login with a web browser)
+# Follow the browser prompts to complete authentication
+```
+
+### Creating a Pull Request with GitHub CLI
+
+```bash
+# 1. Navigate to your repository root
+cd /path/to/your/repository
+
+# 2. Create a Pull Request from your current branch
+gh pr create --title "Fix: Change item name from 'foo' to 'fixme'" --body "## Bug Fix
+- Fixed the test failure where item name was 'foo' instead of expected 'fixme'
+- Updated gilded_rose.py to return correct item name
+- All tests now pass
+
+## Changes Made
+- Modified the item creation logic in gilded_rose.py
+- Ensured test_gilded_rose.py test_foo() passes
+
+## Testing
+- [x] Ran \`python -m unittest\` - all tests pass
+- [x] Verified the fix resolves the failing test"
+
+# 3. Or create a PR interactively (will prompt for title and body)
+gh pr create
+
+# 4. View your PR in the browser
+gh pr view --web
+
+# 5. Check PR status
+gh pr status
+```
+
+### Additional GitHub CLI Commands
+
+```bash
+# List all PRs
+gh pr list
+
+# View a specific PR
+gh pr view [PR_NUMBER]
+
+# Edit a PR
+gh pr edit [PR_NUMBER]
+
+# Merge a PR
+gh pr merge [PR_NUMBER]
+
+# Close a PR
+gh pr close [PR_NUMBER]
+
+# Check out a PR locally
+gh pr checkout [PR_NUMBER]
+```
+
+## Method 3: Direct URL
+You can also create a PR by going directly to:
+
+```
+https://github.com/YOUR_USERNAME/YOUR_REPO/compare/bug/2-bug-foo-instead-of-fixme
+```
+
+## What to Include in Your PR:
+
+### Title:
+```
+Fix: Change item name from 'foo' to 'fixme'
+```
+
+### Description:
+```
+## Bug Fix
+- Fixed the test failure where item name was 'foo' instead of expected 'fixme'
+- Updated gilded_rose.py to return correct item name
+- All tests now pass
+
+## Changes Made
+- Modified the item creation logic in gilded_rose.py
+- Ensured test_gilded_rose.py test_foo() passes
+
+## Testing
+- [x] Ran `python -m unittest` - all tests pass
+- [x] Verified the fix resolves the failing test
+```
+
+## After Creating the PR:
+
+1. Wait for review from team members
+2. Address any feedback by pushing more commits to the same branch
+3. Once approved, merge the PR
+4. Delete the branch after merging (GitHub usually offers this option)
+
+The web interface method is the most user-friendly and gives you a nice visual interface to set up your PR properly!

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -109,6 +109,13 @@ git branch bug/foo-instead-of-fixme
 git checkout bug/foo-instead-of-fixme
 ```
 
+You should see something like:
+
+```
+branch 'bug/2-bug-foo-instead-of-fixme' set up to track 'origin/bug/2-bug-foo-instead-of-fixme'.
+Switched to a new branch 'bug/2-bug-foo-instead-of-fixme'
+```
+
 # 3. Work on your fix while in the virtual environment
 
 See [README.md](./100/README.md)

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -1,4 +1,4 @@
-# 100 - Fixing bug "foobar instead of fixme"
+# 100 - Fixing bug "foo instead of fixme"
 
 When we first ran the unit test, we were alerted about below bug.
 
@@ -89,31 +89,59 @@ In the new issue on GitHub click **Create**.
 
 Choose: ```Create a branch for this issue```.
 
-Accept the proposed name of the new branch (here: ```1-bug-foo-instead-of-fixme```) , choose **Main** as branch source (optionally choose ```Checkout locally```, although we will work from our remote repository) and choose **Create Branch**.
+Modify the proposed name of the new branch (here: ```2-bug-foo-instead-of-fixme```) by prefixing it with in this case ```bug/``` (so the branch name becomes ```bug/2-bug-foo-instead-of-fixme```), choose **Main** as branch source (optionally choose ```Checkout locally```, although we will work from our remote repository) and choose **Create Branch**.
 
+# 2. Switch to your bug specific branch
 
+## Option A: If you created the branch locally
+```bash
+# Check current branch
+git branch
 
+# Switch to your bug branch (if you're not already on it)
+git checkout bug/2-bug-foo-instead-of-fixme
 
-WE ARE HERE 
-
-
-
-So lets create a new branch for the bug.
-
-```
-# 1. Create and switch to a new branch
-git checkout -b bug/foo-instead-of-fixme
-
-# 2. Or alternatively, create the branch first then switch
-git branch bug/foo-instead-of-fixme
-git checkout bug/foo-instead-of-fixme
+# Verify you're on the correct branch
+git status
 ```
 
-You should see something like:
+## Option B: If you created the branch on GitHub (remote)
+```bash
+# Fetch the latest changes from remote
+git fetch origin
 
+# List all available branches (local and remote)
+git branch -a
+
+# Switch to the remote branch (this creates a local tracking branch)
+git checkout -b bug/2-bug-foo-instead-of-fixme origin/bug/2-bug-foo-instead-of-fixme
+
+# Verify you're on the correct branch
+git status
 ```
-branch 'bug/2-bug-foo-instead-of-fixme' set up to track 'origin/bug/2-bug-foo-instead-of-fixme'.
-Switched to a new branch 'bug/2-bug-foo-instead-of-fixme'
+
+## Option C: If the branch already exists remotely
+```bash
+# Fetch all remote branches
+git fetch origin
+
+# Switch to the remote branch
+git checkout bug/2-bug-foo-instead-of-fixme
+
+# If the branch doesn't exist locally, create it and track the remote
+git checkout -b bug/2-bug-foo-instead-of-fixme origin/bug/2-bug-foo-instead-of-fixme
+```
+
+## Verify your setup
+```bash
+# Check current branch
+git branch
+
+# Check remote tracking
+git branch -vv
+
+# You should see something like:
+# * bug/2-bug-foo-instead-of-fixme 1234567 [origin/bug/2-bug-foo-instead-of-fixme] Latest commit message
 ```
 
 # 3. Work on your fix while in the virtual environment

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -232,7 +232,8 @@ After pushing your branch, you can create a Pull Request (PR) in several ways. H
 4. Fill out the PR form:
 - Title: "Fix: Change item name from 'foo' to 'fixme'"
 - Description: Explain what the bug was and how you fixed it
-- Reviewers: Add any team members who should review<br/>
+- Reviewers: Add any team members who should review
+
 5.Click "Create pull request"
 
 ## Method 2: GitHub CLI (if you have it installed)

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -1,5 +1,7 @@
 # 100 - Fixing bug "foo instead of fixme"
 
+# 1. Prepare for bug fixing
+
 When we first ran the unit test, we were alerted about below bug.
 
 ```

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -233,8 +233,7 @@ After pushing your branch, you can create a Pull Request (PR) in several ways. H
 - Title: "Fix: Change item name from 'foo' to 'fixme'"
 - Description: Explain what the bug was and how you fixed it
 - Reviewers: Add any team members who should review
-
-5.Click "Create pull request"
+5. Click "Create pull request"
 
 ## Method 2: GitHub CLI (if you have it installed)
 

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -397,3 +397,16 @@ Fix: Change item name from 'foo' to 'fixme'
 4. Delete the branch after merging (GitHub usually offers this option)
 
 The web interface method is the most user-friendly and gives you a nice visual interface to set up your PR properly!
+
+# 5. Switch back to the main branch
+
+```bash
+# Switch back to the main branch
+git checkout main
+
+# Verify you're on the main branch
+git branch
+
+# Pull any latest changes from remote
+git pull origin main
+```

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -232,7 +232,7 @@ After pushing your branch, you can create a Pull Request (PR) in several ways. H
 4. Fill out the PR form:
 - Title: "Fix: Change item name from 'foo' to 'fixme'"
 - Description: Explain what the bug was and how you fixed it
-- Reviewers: Add any team members who should review
+- Reviewers: Add any team members who should review<br/>
 5.Click "Create pull request"
 
 ## Method 2: GitHub CLI (if you have it installed)

--- a/300/100/200/100/README.md
+++ b/300/100/200/100/README.md
@@ -66,6 +66,71 @@ Your branch is up to date with 'origin/main'.
 nothing to commit, working tree clean
 ```
 
+## Method 1: Local branch creation for bug fixing
+
+### Create a new branch locally
+
+```bash
+# 1. Ensure you're on the main branch and it's up to date
+git checkout main
+git pull origin main
+
+# 2. Create and switch to a new branch for the bug fix
+git checkout -b bug/2-bug-foo-instead-of-fixme
+
+# 3. Verify you're on the new branch
+git branch
+# You should see: * bug/2-bug-foo-instead-of-fixme
+
+# 4. Check the status
+git status
+# Should show: On branch bug/2-bug-foo-instead-of-fixme
+```
+
+### Alternative: Create branch first, then switch
+
+```bash
+# 1. Create the branch without switching to it
+git branch bug/2-bug-foo-instead-of-fixme
+
+# 2. Switch to the new branch
+git checkout bug/2-bug-foo-instead-of-fixme
+
+# 3. Verify you're on the correct branch
+git status
+```
+
+### Push the new branch to remote
+
+```bash
+# 1. Push the new branch to the remote repository
+git push -u origin bug/2-bug-foo-instead-of-fixme
+
+# 2. Verify the branch was pushed
+git branch -vv
+# Should show: * bug/2-bug-foo-instead-of-fixme 1234567 [origin/bug/2-bug-foo-instead-of-fixme] Latest commit message
+```
+
+### Verify your setup
+
+```bash
+# Check current branch
+git branch
+
+# Check remote tracking
+git branch -vv
+
+# Check status
+git status
+
+# You should see:
+# On branch bug/2-bug-foo-instead-of-fixme
+# Your branch is up to date with 'origin/bug/2-bug-foo-instead-of-fixme'.
+# nothing to commit, working tree clean
+```
+
+## Method 2: GitHub Issue Creation (Recommended)
+
 On GitHub create a new issue: ```Bug: foo instead of fixme``` of type **Bug**, with the following description.
 
 ```

--- a/300/100/200/README.md
+++ b/300/100/200/README.md
@@ -43,7 +43,7 @@ This is the intended starting point of the kata! You should now:
 
 Below we will go through each refactoring/bug fixing exercise.
 
-## 100 - Fixing bug "foobar instead of fixme"
+## 100 - Fixing bug "foo instead of fixme"
 
 See [README.md](./100/README.md)
 

--- a/300/100/README.md
+++ b/300/100/README.md
@@ -2,6 +2,8 @@
 
 Based on [Gilded Rose Kata in Python on GitHub](https://github.com/emilybache/GildedRose-Refactoring-Kata/tree/main/python).
 
+**Recommended**: Watch [Python refactoring exercise "GildedRose"](https://www.youtube.com/watch?v=IpauepiaalY).
+
 ## 100 - Instructions
 
 See [README.md](./100/README.md)

--- a/python/tests/test_gilded_rose.py
+++ b/python/tests/test_gilded_rose.py
@@ -9,7 +9,7 @@ class GildedRoseTest(unittest.TestCase):
         items = [Item("foo", 0, 0)]
         gilded_rose = GildedRose(items)
         gilded_rose.update_quality()
-        self.assertEqual("fixme", items[0].name)
+        self.assertEqual("foo", items[0].name) # changed "foo" to "fixme" to fix the test
 
         
 if __name__ == '__main__':

